### PR TITLE
chore(docs): Updates CSS-in-JS docs to use getStyleElement()

### PR DESCRIPTION
**Pull request by jfuchs**
I ran into an error when following the CSS-in-JS styling instructions: using styled components' `getStyleTags()` method returns an HTML string, which was escaped by React as it is currently used.

This change primarily replaces the call to `getStyleTags()` with a call to `getStyleElement()`, which I expect was the intent in these docs? Alternately, I guess you could call `dangerouslySetInnerHTML()` with the output of `getStyleTags()`.

There are also a couple small docs cleanups within the CSS-in-JS seciton.

### CODE REVIEW

File: docs/guides/styling.md [hunk 1]
```diff
@@ -604,23 +604,22 @@ npm add -D concurrently
...
    import { createContext } from "react";
-   export default createContext<null | string>(null);
+   export default createContext<null | React.ReactElement>(
+     null
+   );
    ```
 
 2. Your `entry.server.tsx` will look something like this:
 
-   ```tsx filename=entry.server.tsx lines=6,7,16,20-26,29-30,35,37
```
Comment (jfuchs) [inline docs/guides/styling.md:621]: I couldn't see anywhere the `lines=` values were used in the docs, but I'm happy to bring them back if I'm missing something.

Comment (ryanflorence) [reply docs/guides/styling.md:621]: oops, they're supposed to be inside of `[]` like `lines=[...]`, you mind adding that so it's easier to see where the relevant code changes go?

Comment (jfuchs) [reply docs/guides/styling.md:621]: Great, can do. I couldn't find docs on how to run the docs site locally — is that possible? I'd like to verify that I am counting lines correctly.

Comment (ryanflorence) [reply docs/guides/styling.md:621]: nah, haven't opened up that repo yet, if you're off we'll fix 'em up later :)

Comment (jfuchs) [reply docs/guides/styling.md:621]: Gotcha. Will do.

**Review by mizalewski — APPROVED** [review]

**Comment by ryanflorence** [comment]
I'm sorry we weren't super response over the holidays and as we've gotten the framework (and new team!) off and running. We'll be more responsive moving forward.

I'm going to close this but will gladly (and promptly) merge your updates in a new PR that's made against the current main branch without conflicts :)

**Comment by jfuchs** [comment]
No worries, we all get busy. I don't expect I'll get around to fixing this up any time soon, so if anyone happens across this PR they're welcome to give it another try.

**Pull request closed without merging**